src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -55,7 +55,7 @@ internal static class FatalError
         /// if the host desires to NOT crash the process on a non fatal exception.
         /// </summary>
         [DisallowNull]
-        public static Action<Exception>? NonFatalHandler
+        public static Action<Exception, bool>? NonFatalHandler
```

Comment:
I'd much prefer we start utilizing severity in some way. Faults already report a severity, we just only use "Diagnostic"

Reply:
@ryzngard What do you mean? Happy to make further tweaks here if we can.

Reply:
In the fault creation there is `FaultSeverity.Diagnostic`. Ideally what we do is pass https://sourceroslyn.io/#Microsoft.CodeAnalysis.CodeStyle/LogLevel.cs,13 and then translate that to a `FaultSeverity`. Then we can also use that severity to determine if we capture logs (or even other information). It would also report differently in the telemetry which could be helpful

Reply:
@ryzngard do we want to track that work somewhere? I agree it's a great proposal, but I'm not sure I want to make larger changes here given I'm hoping we take this for servicing.

Reply:
Sure, feel free to file an issue.

Reply:
Filed https://github.com/dotnet/roslyn/issues/57699.